x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/TraceSummary.tsx

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiTitle, EuiText, EuiBadge, EuiToolTip } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { idx } from '@kbn/elastic-idx';
+import moment from 'moment';
+import euiLightVars from '@elastic/eui/dist/eui_theme_light.json';
+import styled from 'styled-components';
+import { Transaction } from '../../../../../typings/es_schemas/ui/Transaction';
+import { asTime, asPercent } from '../../../../utils/formatters';
+import { NOT_AVAILABLE_LABEL } from '../../../../../common/i18n';
+import { isRumAgentName } from '../../../../../common/agent_name';
+import { ErrorCountBadge } from './ErrorCountBadge';
+import { truncate } from '../../../../style/variables';
+
+// TODO: Light/Dark theme (@see https://github.com/elastic/kibana/issues/44840)
+const theme = euiLightVars;
+
+const Heading: React.FunctionComponent = () => (
```

Comment:
I've never understood the appeal of these annotations (but I'm definitely in minority here :D). If we are going to use them what do you think about the shorthand:
```
import React, { FC } from 'react';
const Heading: FC = () => (
```
?

Reply:
I'm fine with using `FC`. The annotations are helpful because a function component is not just a regular function. If you use these annotations you get autocomplete for react-specific properties: 

![image](https://user-images.githubusercontent.com/9912/64351307-b1311d80-cfbf-11e9-8f68-c6502b8fcc29.png)

I like it because it's more explicit and gives the user and the type checker more context.

Reply:
Yeah, I can see that. However, I barely think you can find any usages of those props anywhere. I recall `displayName` was needed for HOC (replaced by hooks),`defaultProps` is replaced with es6 default parameters and `propTypes` is replaced with Typescript.
I can't remember how `contextTypes` - we don't use it even though we use context a lot.

So to me it is a solution to a problem I don't have :)

Reply:
For me it's less about the autocomplete and more about explicitly stating,  "this is what this thing is."

Reply:
Makes sense. I find a big value in Typescript's explicitness . It's like code comments that are guaranteed to be up to date.